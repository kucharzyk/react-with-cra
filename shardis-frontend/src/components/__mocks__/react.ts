import * as React from 'react';

// Temporary hack to suppress error
// https://github.com/facebookincubator/create-react-app/issues/3199
window.requestAnimationFrame = function (callback) {
  setTimeout(callback, 0);
  return 0;
};

module.exports = React;
