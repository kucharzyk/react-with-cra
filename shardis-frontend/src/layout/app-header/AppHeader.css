.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 48px;
  margin-top: 8px;
}

.App-header h2 {
  color: white;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
