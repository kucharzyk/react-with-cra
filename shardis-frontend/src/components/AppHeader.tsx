import * as React from 'react';
import { StatelessComponent } from 'react';
import './AppHeader.css';

const logo = require('./logo.svg');

const AppHeader: StatelessComponent = () => (
  <div className="App-header">
    <img src={logo} className="App-logo" alt="logo"/>
    <h2>Welcome to React</h2>
  </div>
);

export default AppHeader;
