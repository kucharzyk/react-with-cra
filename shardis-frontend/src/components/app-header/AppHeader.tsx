import * as React from 'react';
import { StatelessComponent } from 'react';

import './AppHeader.css';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import Tabs, { Tab } from 'material-ui/Tabs';
import { RouteComponentProps, withRouter } from 'react-router';
import Hidden from 'material-ui/Hidden';
import { StyleRulesCallback, Theme } from 'material-ui/styles';
import withStyles from 'material-ui/styles/withStyles';
import { WithStyles } from 'material-ui';

const logo = require('./logo.svg');

const styles: StyleRulesCallback<'root'> = (theme: Theme) => ({
  root: {},
});

const getActiveMenuIndex = (pathname: string) => {
  if (pathname.startsWith('/404')) {
    return 4;
  } else if (pathname.startsWith('/time')) {
    return 3;
  } else if (pathname.startsWith('/about')) {
    return 2;
  } else if (pathname.startsWith('/home')) {
    return 1;
  } else {
    return 0;
  }
};

const AppHeader: StatelessComponent<RouteComponentProps<{}> & WithStyles<'root'>> = (props) => (
  <div className="App-header">
    <AppBar position="static" color="default">
      <Toolbar>
        <img src={logo} className="App-logo" alt="logo"/>
        <Typography type="title" color="inherit">
          <span className="app-title">Welcome to React</span>
        </Typography>
        <Hidden smDown={true}>
          <Tabs value={getActiveMenuIndex(props.location.pathname)} style={{marginLeft: '1em'}}>
            <Tab label="Main Page" onClick={() => props.history.push('/')}/>
            <Tab label="Home" onClick={() => props.history.push('/home')}/>
            <Tab label="About" onClick={() => props.history.push('/about')}/>
            <Tab label="Time" onClick={() => props.history.push('/time')}/>
            <Tab label="404" onClick={() => props.history.push('/404')}/>
          </Tabs>
        </Hidden>
      </Toolbar>
    </AppBar>
  </div>
);

export default withRouter(withStyles(styles)(AppHeader));
