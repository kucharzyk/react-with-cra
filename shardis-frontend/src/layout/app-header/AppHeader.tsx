import * as React from 'react';
import { StatelessComponent } from 'react';
import { Layout } from 'antd';

import './AppHeader.css';

const logo = require('./logo.svg');

const AppHeader: StatelessComponent = () => (
  <Layout.Header style={{padding: 0}}>
    <div className="App-header">
      <img src={logo} className="App-logo" alt="logo" style={{float: 'left'}}/>
      <h2 style={{float: 'left'}}>Welcome to React</h2>
      <div style={{clear: 'both'}}/>
    </div>
  </Layout.Header>
);

export default AppHeader;
