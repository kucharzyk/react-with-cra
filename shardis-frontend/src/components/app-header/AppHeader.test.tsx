import * as React from 'react';
import { StatelessComponent } from 'react';
import * as enzyme from 'enzyme';

import AppHeader from './AppHeader';
import { StaticRouter } from 'react-router';
import toJson from 'enzyme-to-json';
import createShallow from 'material-ui/test-utils/createShallow';

describe('AppHeader', () => {

  let WrappedComponent: StatelessComponent;

  beforeEach(() => {
    WrappedComponent = () => (
      <StaticRouter location={'/'} context={{}}>
        <AppHeader/>
      </StaticRouter>
    );
  });

  it('renders snapshot', () => {
    const shallowComponent = createShallow({untilSelector: 'AppHeader'})(<WrappedComponent/>);
    expect(toJson(shallowComponent)).toMatchSnapshot();
  });

  it('renders the correct text', () => {
    const component = enzyme.render(<WrappedComponent/>);
    expect(component.find('.app-title').text()).toEqual('Welcome to React');
  });

});
