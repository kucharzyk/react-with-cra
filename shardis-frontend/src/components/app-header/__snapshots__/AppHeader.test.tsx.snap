// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader renders snapshot 1`] = `
<div
  className="App-header"
>
  <WithStyles(AppBar)
    color="default"
    position="static"
  >
    <WithStyles(Toolbar)>
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <WithStyles(Typography)
        color="inherit"
        type="title"
      >
        <span
          className="app-title"
        >
          Welcome to React
        </span>
      </WithStyles(Typography)>
      <Hidden
        implementation="js"
        lgDown={false}
        lgUp={false}
        mdDown={false}
        mdUp={false}
        smDown={true}
        smUp={false}
        xlDown={false}
        xlUp={false}
        xsDown={false}
        xsUp={false}
      >
        <WithStyles(Tabs)
          style={
            Object {
              "marginLeft": "1em",
            }
          }
          value={0}
        >
          <WithStyles(Tab)
            label="Main Page"
            onClick={[Function]}
          />
          <WithStyles(Tab)
            label="Home"
            onClick={[Function]}
          />
          <WithStyles(Tab)
            label="About"
            onClick={[Function]}
          />
          <WithStyles(Tab)
            label="Time"
            onClick={[Function]}
          />
          <WithStyles(Tab)
            label="404"
            onClick={[Function]}
          />
        </WithStyles(Tabs)>
      </Hidden>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;
